pkg/network/link/discovery_test.go

```diff
@@ -0,0 +1,116 @@
+/*
+ * This file is part of the KubeVirt project
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ * Copyright 2023 Red Hat, Inc.
+ *
+ */
+
+package link
+
+import (
+	"errors"
+
+	"github.com/golang/mock/gomock"
+	. "github.com/onsi/ginkgo/v2"
+	. "github.com/onsi/gomega"
+
+	"github.com/vishvananda/netlink"
+
+	v1 "kubevirt.io/api/core/v1"
+
+	netdriver "kubevirt.io/kubevirt/pkg/network/driver"
+)
+
+var _ = Describe("DiscoverByNetwork", func() {
+	const (
+		testNetworkName         = "blue"
+		testNetIfaceName        = "pod16477688c0e"
```

Comment:
why are we using the `pod` prefix for the pod interfaces ?

Reply:
It was my request. In order to be consistent with all the other devices created.
It also makes it easier to process and identify the pod interface.
In the past I think I asked to remove it, but now it was clear from the follow up changes that it is a mess without the embedded type identifier.

I hope a follow up PR will make the <type><hash> format for all (currently the bridge and dummy have quirks).